src/main/java/com/softserve/service/impl/SemesterServiceImpl.java

```diff
@@ -301,49 +326,178 @@ public Semester addGroupToSemester(Semester semester, Group group) {
+        getById(semester.getId()).setPeriods(periodsSemester);
+        return semester;
+    }
+
+    /**
      * Method delete group from an existing semester
      *
      * @param semester semester in which we need to delete group
-     * @param group group to delete
+     * @param group    group to delete
      * @return changed Semester
      */
     @Override
     public Semester deleteGroupFromSemester(Semester semester, Group group) {
-        log.info("In deleteGroupFromSemester (semester = [{}], group = [{}])", semester, group);
+        log.debug("In deleteGroupFromSemester (semester = [{}], group = [{}])", semester, group);
         List<Group> groups = semester.getGroups();
         groups.remove(group);
         update(semester);
         return semester;
     }
 
     /**
+     * Method delete groups, periods and days Of Weeks from an existing semester
      *
+     * @param semester semester in which we need to delete group
+     * @return changed Semester
+     */
+    @Override
+    public Semester deleteAllContentFromSemester(Semester semester) {
+        log.debug("In deleteAllContentFromSemester (semester = [{}] )", semester);
+        List<Group> groups = new ArrayList<>();
+        Set<Period> periods = new HashSet<>();
+        Set<DayOfWeek> dayOfWeeks = new HashSet<>();
+        semester.setGroups(groups);
+        semester.setPeriods(periods);
+        semester.setDaysOfWeek(dayOfWeeks);
+        update(semester);
+        return semester;
+    }
+
+    /**
      * Method delete groups from an existing semester
      *
      * @param semester semester in which we need to delete groups
-     * @param groups group to delete
+     * @param groups   group to delete
      * @return changed Semester
      */
     @Override
     public Semester deleteGroupsFromSemester(Semester semester, List<Group> groups) {
-        log.info("In deleteGroupsFromSemester (semester = [{}], group = [{}])", semester, groups);
+        log.debug("In deleteGroupsFromSemester (semester = [{}], group = [{}])", semester, groups);
         groups.forEach(group -> deleteGroupFromSemester(semester, group));
         return semester;
     }
+
+    /**
+     * Method copy groups, periods, days Of Weeks and Schedule from one to other semester
+     *
+     * @param fromSemesterId id semester from which we need to copy groups, periods, days Of Weeks and Schedule
+     * @param toSemesterId   id semester in which we need to copy groups, periods, days Of Weeks and Schedule
+     * @return copied Semester
+     */
+    @Override
+    public Semester copySemester(Long fromSemesterId, Long toSemesterId) {
+
+        Semester toSemester = getById(toSemesterId);
+        Semester fromSemester = getById(fromSemesterId);
+        List<Schedule> schedules = scheduleRepository.getScheduleBySemester(fromSemesterId);
+        for (Schedule schedule : schedules) {
+            Hibernate.initialize(schedule.getLesson().getSemester().getDaysOfWeek());
```

Comment:
please, fetch always with data that you need. Use JOIN FETCH

Reply:
The Hibernate.initialize approach is applied throughout the project, example in LessonServiceImpl and ScheduleServiceImpl.
Maybe it's worth creating a separate task to fix the whole project?

Reply:
Done